web-client/app/scripts/services/linkService.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+angular.module('webClientApp')
+  .service('Link', ['$resource', '$http', '$q', 'API_HOST',
+      function ($resource, $http, $q, API_HOST) {
+
+      var baseUrl = '//' + API_HOST + '/api/v1/';
+      var LinkResource = $resource(baseUrl + 'links/:linkId');
+
+      /**
+       * These configs are needed. AngularJS identity tranformer
+       * can automatically figure out that this is a multipart
+       * content and use the correct Content-Type.
+       */
+      var configs = {
```

Comment:
This is not necessary here. The config, createFormData and the custom save and update methods are only needed for   services that needs to upload files. Drop all of it and just return the resource itself